### ISSUE TITLE
Duplicate generated annotation classes in Eclipse

### ISSUE THREAD

**Issue by ChristianCiach** [tbroyer/gradle-apt-plugin#85]
Since updating the plugin to 0.16, the generated .classpath file for Eclipse often contains a wrong source-directory entry, resulting in duplicated generated classes in Eclipse.

We use Gradle 4.8 and Eclipse 4.7.3a. I use Buildship 3.0.0.v20180607, but my colleagues have this issue without using Buildship (just using "gradlew eclipse" on the command line).

I don't know the exact steps to reproduce this problem, but these steps often work:

1. Have an Eclipse project with annotation processing enabled (after executing the tasks "eclipseFactorypath" and "eclipseJdtApt").
2. Everything works, because the generated classes are present under ".apt_generated" folder.
3. Build the project on the command line (doesn't always seem necessary for reproducing the issue) by using "gradlew compileJava".
4. Use "Refresh Gradle project" in the context menu of the Eclipse project.
5. Result: There is an added source folder with the name "build/generated/source/apt/main", containing the same files as ".apt_generated".

The wrongly added source folder can be seen inside the .classpath file (third classpathentry):

```
<?xml version="1.0" encoding="UTF-8"?>
<classpath>
	<classpathentry kind="src" output="bin/main" path="src">
		<attributes>
			<attribute name="gradle_scope" value="main"/>
			<attribute name="gradle_used_by_scope" value="main,test"/>
		</attributes>
	</classpathentry>
	<classpathentry kind="src" output="bin/test" path="test">
		<attributes>
			<attribute name="gradle_scope" value="test"/>
			<attribute name="gradle_used_by_scope" value="test"/>
		</attributes>
	</classpathentry>
	<classpathentry kind="src" output="bin/main" path="build/generated/source/apt/main">
		<attributes>
			<attribute name="gradle_scope" value="main"/>
			<attribute name="gradle_used_by_scope" value="main,test"/>
		</attributes>
	</classpathentry>
	<classpathentry kind="con" path="org.eclipse.buildship.core.gradleclasspathcontainer">
		<attributes>
			<attribute name="annotationpath" value="/MyProject/annotations"/>
		</attributes>
	</classpathentry>
	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER"/>
	<classpathentry kind="src" path=".apt_generated">
		<attributes>
			<attribute name="optional" value="true"/>
		</attributes>
	</classpathentry>
	<classpathentry kind="output" path="bin/default"/>
</classpath>
```

Downgrading to 0.15 solves this issue.

**Comment by tbroyer** [comment]
You might try excluding the `build/generated/source/apt` folder (and/or subfolders) using a `resourceFilter`: https://docs.gradle.org/current/dsl/org.gradle.plugins.ide.eclipse.model.EclipseProject.html

If that doesn't work, then just stick to 0.15 for now :man_shrugging: (besides internal cleanup, adding the generated sources directories to the source sets' `allSource` and `allJava` is the real change in 0.16)

**Comment by ChristianCiach** [comment]
Thanks! We will just stick to 0.15 for now.

**Comment by tbroyer** [comment]
Fwiw, I reverted that change and released v0.17 (which is, functionally, the same as v0.15)
